presto-hive/src/main/java/io/prestosql/plugin/hive/DynamicConfigurationUpdater.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import org.apache.hadoop.conf.Configuration;
+
+import java.net.URI;
+
+public interface DynamicConfigurationUpdater
+{
+    Configuration updateConfiguration(Configuration configuration, HdfsEnvironment.HdfsContext context, URI uri);
```

Comment:
Nit: add a direct import for `HdfsContext` (so it doesn't need to be prefixed with `HdfsEnvironment`)